### ISSUE TITLE
Engine is returning reversed order of events for creating a new standalone task

### ISSUE THREAD

**Issue by lucianoprea** [Activiti/Activiti#1854]
If a new standalone task is created with an assignee using the `TaskService` the engine triggers the **TASK_ASSIGNED** event before **TASK_CREATED**. 
Investigate the impact of changing this inside the engine. 
@balsarori provided some context:

> - new task is created in UserTaskActivityBehavior.java, insert method is called here
> https://github.com/Activiti/Activiti/blob/develop/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/UserTaskActivityBehavior.java#L196
> 
> - which fires TASK_ASSIGNED is here
> https://github.com/Activiti/Activiti/blob/develop/activiti-engine/src/main/java/org/activiti/engine/impl/persistence/entity/TaskEntityManagerImpl.java#L99
> - later in UserTaskActivityBehavior.java the TASK_CREATED event is triggered
> https://github.com/Activiti/Activiti/blob/develop/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/UserTaskActivityBehavior.java#L216
> - should be possible to do an insert without triggering TASK_ASSIGNED and later fire TASK_CREATED & TASK_ASSIGNED events

**Comment by ryandawsonuk** [comment]
While this is an issue we should careful with it as it's by design. Activiti in action notes it was done to allow task listeners to change the assignee without getting into an infinite loop.

**Comment by Salaboy** [comment]
Yeah, Can we close this? or is this still an issue? Can we handle these at the services layer?

**Comment by balsarori** [comment]
Workarounds can be done by consumers of the events, however it's better to fix it in the engine. The order of events is not correct which will confuse users. I don't think it's hard to fix it in the engine. If it's not fixed in the engine now it will be harder to fix later when there are lots of consumers using them in that incorrect order.

**Comment by Salaboy** [comment]
@lucianoprea can you please point me to where we are doing that..

**Comment by lucianoprea** [comment]
@Salaboy for creating a new standalone task in RB I had to do this [workaround](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/blob/baeb4f003dd60266c9ed0b3b2944340b06e89fed/activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/ProcessEngineWrapper.java#L216).
Also I've added a test that validates that the order of events returned by engine is reversed. Check it  [here](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/blob/6bb5e2a0a845ec9b0f3bb20395b6e755a36f66d9/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/test/java/org/activiti/cloud/services/events/listeners/MessageProducerActivitiEventListenerIT.java#L115).
